SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -35,41 +35,152 @@ Incorporating the required abstractions to support payment modules will require
-For any preference list, the chosen option will be the option which has the highest cumulative preference. The preference list has three dimensions, which will be resolved from high to low:
+There are three dimensions to take into consideration for the payment preferences: 
 
 1. Currency to use
 2. Price oracle to use
 3. Payment method to use
 
-Nodes must be able to reach an agreement in any dimension of the preference list, or the fallback option for each dimension of the preference list will be chosen:
+Nodes should be allowed to specify their payment preferences and indicate which dimension (or dimensions) will have the biggest impact when negotiating the payment mechanism to use with their peers.
+
+The following is an example of the proposed configuration:
+
+```yaml
+# How important a dimension is for the node at the moment of negotiating with a peer the payment mechanism to use
+dimensions:
+	currency: 70
+	provider: 20
+	oracle: 10
+
+# Supported payment options by the node.
+# A payment is defined by a currency, the weight (preference) for such currency and the preferred providers and oracles.
+# Provider preference for a given currency is defined by the order of the provider. As an example, for the rif currency Lumino has a  
+# higher preference over Raiden. 
+payments:
+	rif:
+		weight: 36
+		providers:
+			lumino:
+				weigth: 15
+			raiden:
+				weight: 45
+		oracles:
+			oracleA:
+				weigth: 10
+			oracleB:
+				weigth: 70
+			oracleC:
+				weigth: 60
+	xdai:
+		weight: 9
+		providers:
+			Lumino: 
+				weigth: 10
+		oracles:
+			oracleA: 
+				weight: 20
+	dai:
+		weight: 5
+		providers:
+			raiden:
+				weight: 80
+		oracles:
+			oracleC: 
+				weight: 30
+			oracleE:
+				weight: 20
+	eth:
+		weight: 40
+		providers:
+			swap:
+				weight: 35
+		oracles:
+			oracleB:
+				weight: 90
+			oracleE:
+				weight: 10
+```
+
+It's important to mention that weights can be any numeric value and that they are normalized.
+
+The selection algorithm during handshake works like this:
+
+1. Based on the node configuration generate triplets of the form ```[currency, provider, oracle]``` and assign to the triplet the corresponding weight from the config file.
+2. Exchange the generated triplets with the peer.
+3. Each peer will eliminate triplets they don't have in common.
+4. Compute the weighted preference of each remaining triplet from node A and node B as:
+
+	```golang
+	weightedPreference := normalized(dimensions.coin) * normalized(triplet[n].weight) + 
+						normalized(dimensions.provider) * normalized(triplet[n].provider.weight) + 
+						normalized(dimensions.oracle) * normalized(triplet[n].oracle.weight)
+	```
+	
+	as an example let's take from the previous configuration the ```rif``` entry. From that entry we can build the triplet [rif, lumino, oracleB] with an assigned weight of 36. The coin, provider and oracle dimensions are 70, 20 and 10 respectively, Lumino has a weight of 15 and oracleB has a weight of 70. 
```

Comment:
I think it will be more clear and concise if you delete anything after [rif, lumino, oracleB], as you explain it below as well but in more detail.